File: src/proxy/examples/owneranonymizer/RealOwnerContact.js
'use strict';

const OwnerContact = require('./OwnerContact');

class RealOwnerContact extends OwnerContact {
  constructor(owner) {
    super();
    this.owner = owner;
  }
  getName()  { return this.owner.fullName; }
  getPhone() { return this.owner.phone; }
  getEmail() { return this.owner.email; }
}

module.exports = RealOwnerContact;

File: src/proxy/examples/owneranonymizer/RealOwnerContact.java
package proxy.examples.owneranonymizer;

import petapp.domain.Owner;

public class RealOwnerContact implements OwnerContact {

    private final Owner owner;

    public RealOwnerContact(Owner owner) { this.owner = owner; }

    @Override public String getName()  { return owner.getFullName(); }
    @Override public String getPhone() { return owner.getPhone(); }
    @Override public String getEmail() { return owner.getEmail(); }
}
